src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Publish.targets

```
@@ -694,6 +694,40 @@ Copyright (c) .NET Foundation. All rights reserved.
     </ItemGroup>
   </Target>
 
+  <!--
+    ============================================================
+                                        _CreateSingleFileHost
+      Create the single-file host for the publish scenario if app is being published as a self-contained single-file .net5+ app
+    ============================================================
+     -->
+  <UsingTask TaskName="CreateAppHost" AssemblyFile="$(MicrosoftNETBuildTasksAssembly)" />
+  <Target Name="_CreateSingleFileHost"
+          Inputs="@(IntermediateAssembly);$(SingleFileHostSourcePath)"
+          Outputs="$(SingleFileHostIntermediatePath)"
+          DependsOnTargets="_GetAppHostPaths"
+          Condition="'$(PublishSingleFile)' == 'true' and
+                     '$(SelfContained)' == 'true' and
+                     '$(SingleFileHostSourcePath)' != '' and
+                     $([MSBuild]::IsTargetFrameworkCompatible('$(TargetFramework)', 'net5.0')) and
```

We should use the `TargetFrameworkIdentifier` and `TargetFrameworkVersion` properties for comparisons like this in order to support [target framework aliases](https://github.com/NuGet/Home/issues/5154)

```suggestion
                     '$(TargetFrameworkIdentifier)' == '.NETCoreApp' and
                     $([MSBuild]::VersionGreaterThanOrEquals($(TargetFrameworkVersion), 5.0)) and
```